### ISSUE TITLE
Stuck on Initialising

### ISSUE THREAD

**Issue by iamasink** [VirtCode/SmartMouse#18]
This project seems really cool!  
However the app gets stuck on "Initializing" on my phone, I tried using using SensorServer and it says `Logged In successfully` but receives no data after in the csv.   
I tried resetting/reinstalling the app but just can't seem to get it to work. Am I doing something wrong?  

![Screenshot_20240504-192330189](https://github.com/VirtCode/SmartMouse/assets/35533595/ddae3715-8398-4698-ac5d-78a193c602c6)

**Comment by VirtCode** [comment]
Thanks for trying it out! This looks like something is going wrong while initializing bluetooth. 

Normally, this app uses the Bluetooth HID profile to connect and transmit movement data to a computer. Such a connection is still required when doing debugging (i.e. capturing data with SensorServer) so it won't transmit data until Bluetooth is initialized (which is what seems to be failing) and such a connection is established.

So could you quickly check whether your phone supports the Bluetooth HID profile? This app *should* check for support and show a message otherwise, but maybe that's broken. You can easily check your support with apps like [this](https://play.google.com/store/apps/details?id=com.rdapps.bluetoothhidtester). Afterwards we'll have to check whether we find something in the logs.

**Comment by iamasink** [comment]
Ah, unfortunately I get "No Bluetooth HID Profile Detected!" on that app. I imagine there's no way to get this to work then? I have an ASUS Zenfone 8.

**Comment by VirtCode** [comment]
I'm afraid not. Some vendors have a toggle for the HID profile in the developer settings, but I couldn't find whether that's the case for the Zenfone. Maybe they could ship an update which enables it but realistically that won't happen.

Using the app without the HID profile also isn't really possible. I've specifically opted for this profile because of its ease of use. Otherwise you'd have to have an additional app/server on the target PC, that supports all/most platforms with acceptable UX, and you have to be in the same network, etc. The SensorServer is really only intended for making captures for debugging.

I'll label this as a bug, as this app should've shown "Unsupported" instead of "Initializing".

**Comment by iamasink** [comment]
Yeah I had a look for any option like that but couldn't find anything. Thanks for the help anyways.
